pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/codearea/syntaxhighlighting/JavaSyntaxHighlighter.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.util.codearea.syntaxhighlighting;
+
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import net.sourceforge.pmd.util.fxdesigner.util.codearea.SimpleRegexSyntaxHighlighter;
+
+/**
+ * Syntax highlighter for Java.
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class JavaSyntaxHighlighter extends SimpleRegexSyntaxHighlighter {
+
+
+    private static final String[] KEYWORDS = new String[] {
+        "public", "return", "final", "import", "static", "new",
+        "extends", "int", "throws?", "void", "if", "this",
+        "private", "class", "else", "case", "package", "abstract",
+        "boolean", "break", "byte", "catch", "char", "for",
+        "continue", "default", "double", "enum", "finally",
+        "float", "implements", "instanceof", "interface", "long",
+        "native", "protected", "while", "assert", "short", "super",
```

Comment:
Java 9 added `open`, `module`, `requires`, `transitive`, `exports`, `opens`, `to`, `uses`, `provides`, and `with` as restricted keywords (for backwards compatibility), we should probably include them here.

Also note, that both `const` and `_` are technically keywords.

https://docs.oracle.com/javase/specs/jls/se9/html/jls-3.html#jls-3.9